File: Back-End/db/seeds.rb
# db/seeds.rb

# Clear existing data first (optional)
User.destroy_all
Product.destroy_all

puts "Creating fake users..."
10.times do
  User.create!(
    user_id: Faker::Number.id,
    first_name: Faker::Name.name,
    last_name: Faker::Name.name,
    username: Faker::Name.user,
    password: 'password',
    password_confirmation: 'password',
    email: Faker::Internet.email  
  )
end

users = User.all
puts "Creating fake products..."
20.times do
  Product.create!(
    muse_id: Faker::Number.id,
    name: Faker::Name.name,
    origin: Faker::Name.name,
    media_type: Faker::Media.type,
    img_url: 'url',
    user_id: Faker::Number.id
  )
end

puts "Database seeded successfully with #{User.count} users and #{Product.count} products!"

File: Back-End/db/seeds.js
import { faker } from "@faker-js/faker";
//import db from client.js
import db from "#db/client";
//db query functions
import { createPlant } from "./queries/plants.js";
import { createUser } from "./queries/users.js";
//plantSpecies for seeding
import plantSpecies from "./plantSpecies.js";

//connect to the database using pg
await db.connect();

//call seedBooks function defined below
await seedTables();

//end the connection to the database
await db.end();

//log a message to the console ensuring a successing seeding
console.log("🌱 Database seeded.");

async function seedTables() {
  for (let i = 0; i < 10; i++) {
    const person = {
      first_name: faker.person.firstName(),
      last_name: faker.person.lastName(),
      password: faker.internet.password(),
      username: faker.internet.username(),
      email: faker.internet.email(),
    };

    const user = await createUser(person);

    const randomNum = Math.round(Math.random());

    const plant = {
      water_frequency: `${Math.floor(Math.random() * 10) + 1} per ${
        randomNum === 0 ? "day" : "week"
      }`,
      species: plantSpecies[Math.floor(Math.random() * plantSpecies.length)],
      user_id: user.user_id,
    };

    await createPlant(plant);
  }
}